package-res/resources/web/pentaho/type/filter/_core/tree.js

```diff
@@ -0,0 +1,201 @@
+/*!
+ * Copyright 2010 - 2016 Pentaho Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+define([
+  "../../../util/error"
+], function(error) {
+
+  "use strict";
+
+  return function(filter) {
+
+    /**
+     * @name pentaho.type.filter.Tree.Type
+     * @class
+     * @extends pentaho.type.Abstract.Type
+     *
+     * @classDesc The base type class of `Tree` filter types.
+     *
+     * For more information see {@link pentaho.type.filter.Tree}.
+     */
+
+    /**
+     * @name pentaho.type.filter.Tree
+     * @class
+     * @extends pentaho.type.filter.Abstract
+     * @abstract
+     *
+     * @amd {pentaho.type.Factory<pentaho.type.filter.Tree>} pentaho/type/filter/tree
+     *
+     * @classDesc The base class of filters that combine other filters.
+     *
+     * Example subclasses include:
+     *
+     * * {@link pentaho.type.filter.And}
+     * * {@link pentaho.type.filter.Or}
+     *
+     * @description Creates a filter that combines a series of filters.
+     *
+     * @constructor
+     * @param {pentaho.type.filter.spec.ITree} [spec] - A tree filter specification.
+     */
+
+    filter.Tree = filter.Abstract.extend("pentaho.type.filter.Tree", /** @lends pentaho.type.filter.Tree# */{
+
+      /**
+       * Gets the list of operands of this filter.
+       *
+       * This getter is a shorthand for `this.get("operands")`.
+       *
+       * @type {pentaho.type.List<pentaho.type.filter.Abstract>}
+       *
+       * @readonly
+       */
+      get operands() {
+        return this.get("operands");
+      },
+
+      /**
+       * Gets the inverse filter class.
+       *
+       * @name _inverseClass
+       * @memberOf pentaho.type.filter.Tree#
+       * @type {Class.<pentaho.type.filter.Tree>}
+       * @protected
+       * @abstract
+       * @readOnly
+       */
+
```

Comment:
Actually, that's a virtual doclet and is not related to the following method, whose name is `negate`.